addons/test_mail/static/tests/systray_activity_menu_tests.js

```diff
@@ -76,104 +82,64 @@ QUnit.test("activity menu widget: activity menu with 2 models", async (assert) =
     await contains(".o_menu_systray i[aria-label='Activities']");
     await contains(".o-mail-ActivityMenu-counter");
     await contains(".o-mail-ActivityMenu-counter", { text: "5" });
-    let context = {};
+    const actionChecks = {
+        context: { force_search_count: 1 },
+        domain: [["activity_user_id", "=", session.uid]],
+    }
     patchWithCleanup(env.services.action, {
         doAction(action) {
-            assert.deepEqual(action.context, context);
+            Object.entries(actionChecks).forEach(([key, value]) => {
+                assert.deepEqual(action[key], value);
+            });
+            assert.step("do_action:" + action.name);
         },
     });
-    context = {
-        force_search_count: 1,
-        search_default_activities_overdue: 1,
-    };
     await click(".o_menu_systray i[aria-label='Activities']");
     await contains(".o-mail-ActivityMenu");
     await contains(".o-mail-ActivityMenu .o-mail-ActivityGroup", { count: 2 });
-    await click(".o-mail-ActivityMenu .o-mail-ActivityGroup button", { text: "1 Late" });
-    await contains(".o-mail-ActivityMenu", { count: 0 });
-    context = {
-        force_search_count: 1,
-        search_default_activities_today: 1,
-    };
-    await click(".o_menu_systray .dropdown-toggle:has(i[aria-label='Activities'])");
-    await click(":nth-child(1 of .o-mail-ActivityGroup) button", { text: "1 Today" });
+    assert.ok(
+        selectContaining(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", [
+            "res.partner",
+            "0 Late",
+        ])
```

Comment:
We don't need to take that much vertical space, especially for tests ^^
```suggestion
        selectContaining(document, ".o-mail-ActivityMenu .o-mail-ActivityGroup", ["res.partner", "0 Late"])
```